app/lib/database/views/flags_view.js

```diff
@@ -12,35 +12,41 @@
  * or submit itself to any jurisdiction.
  */
 
-const flags_view = (query) => {
-    
-    rn = query.run_numbers;
-    let rn_sql = "";
-    if (typeof(rn) === 'object') {
-        rn_sql = rn.join(",");
-    } else if (typeof(rn) === 'string') {
-        rn_sql = rn
+
+const handleArray = (data, field, escape) => {
+    let sqlLogicClause = undefined;
+    if (Array.isArray(data)) {
+        sqlLogicClause = data.map((d) => escape ? `'${d}'` : d).join(',');
+    } else if (typeof(data) === 'string' || !data) {
+        sqlLogicClause = escape ? `'${data}'` : data;
     } else {
-        throw `run_numbers seems to be incorrect ${rn}`
+        throw `incorrect format <${data}> for ${field}`;
     }
+    sqlLogicClause = sqlLogicClause ? `${field} in (${sqlLogicClause})` : undefined;
 
-    let det_sql = "";
-    if (query.detector) {
-        det_sql = `AND ds.name = '${query.detector}'`
-    }
+    return sqlLogicClause;
+}
+
+const flags_view = (query) => {
+    run_selection_sql = handleArray(query.run_numbers, 'r.run_number')
+    detector_selection_sql = handleArray(query.detector, 'ds.name', true)
```

Comment:
Why escaping only the second one?

Reply:
no good name for argument, it indicated whether some value should be treaded as string or not, I changed the name of the argument to 'asString';